src/Access/EntityBasePermissions.php

```diff
@@ -0,0 +1,184 @@
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity type bundle info.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeBundleInfoInterface
+   */
+  protected $entityTypeBundleInfo;
+
+  /**
+   * Creates Drupal\entity\Entity\Access\EntityBasePermissions.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
+   *   The entity type bundle info.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityTypeBundleInfo = $entity_type_bundle_info;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_type.bundle.info')
+    );
+  }
+
+  /**
+   * Gets an array of entity type permissions.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityTypeInterface $entity_type
+   *   The custom entity definition.
+   *
+   * @return array
+   *   The entity type permissions.
+   *
+   * @see \Drupal\user\PermissionHandlerInterface::getPermissions()
+   */
+  public function entityPermissions(ContentEntityTypeInterface $entity_type) {
+    $perms = [];
+
+    if (!empty($entity_type)) {
+      // Get the entity ID.
+      $entity_type_id = $entity_type->id();
+      // Build replacement data for lables and descriptions.
+      $replacements = [
+        '@entity_type_id' => $entity_type_id,
+        '@entity_label' => $entity_type->getLabel(),
+      ];
+      // Add the default entity permissions.
+      $perms = [
+        "bypass $entity_type_id access" => [
+          'title' => $this->t('Bypass @entity_label access control', $replacements),
+          'description' => $this->t('View, edit and delete all @entity_label regardless of permission restrictions.', $replacements),
+          'restrict access' => TRUE,
+        ],
+        "administer $entity_type_id types" => [
+          'title' => $this->t('Administer @entity_label types', $replacements),
+          'description' => $this->t('Promote, change ownership, edit revisions, and perform other tasks across all @entity_label types.', $replacements),
+          'restrict access' => TRUE,
+        ],
+        "administer $entity_type_id" => [
+          'title' => $this->t('Administer @entity_label', $replacements),
+          'restrict access' => TRUE,
+        ],
+        "access $entity_type_id overview" => [
+          'title' => $this->t('Access the @entity_label overview page', $replacements),
+          'description' => $this->t('Get an overview of all @entity_label.', $replacements),
+        ],
+        "access $entity_type_id" => [
+          'title' => $this->t('View published @entity_label', $replacements),
```

Comment:
publishing is not necessarily a concept/wording used, sometimes it's just active/inactive. Would it make sense to use a more generic wording here?